cli/src/commands/extensions/extension.rs

```diff
@@ -0,0 +1,143 @@
+use std::{convert::TryFrom, fs::File, io::Read, path::PathBuf};
+
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+
+const MANIFEST_NAME: &str = "PhylumExt.toml";
+
+#[derive(Debug)]
+pub struct Extension {
+    path: PathBuf,
+    manifest: ExtensionManifest,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ExtensionManifest {
+    name: String,
+    description: Option<String>,
+    entry_point: String,
+}
+
+impl Extension {
+    pub fn name(&self) -> &str {
+        &self.manifest.name
+    }
+
+    pub fn description(&self) -> Option<&str> {
+        self.manifest.description.as_deref()
+    }
+
+    /// Install the extension in the default path.
+    pub fn install(&self) -> Result<()> {
+        println!("Installing extension {}...", self.name());
+        let target_prefix = extensions_path()?.join(self.name());
+
+        if target_prefix == self.path {
+            return Err(anyhow!(
+                "extension path and installation path are identical, skipping"
+            ));
+        }
+
+        for entry in walkdir::WalkDir::new(&self.path) {
+            let source_path = entry?.into_path();
+            let dest_path = target_prefix.join(source_path.strip_prefix(&self.path)?);
+
+            if source_path.is_dir() {
+                std::fs::create_dir_all(dest_path)?;
+            } else if source_path.is_file() {
+                if dest_path.exists() {
+                    return Err(anyhow!("{}: already exists", dest_path.to_string_lossy()));
+                } else {
+                    std::fs::copy(source_path, dest_path)?;
+                }
+            }
+        }
+
+        println!("Extension {} installed correctly", self.name());
```

Comment:
I think the wording "correctly" is a bit strange? As a user I'd usually expect "successfully", since the command should never really *incorrectly* install an extension.